tests/kokkos/kokkos_utils_tests.cpp

```diff
@@ -244,4 +249,96 @@ TEST_CASE("parallel_reduce", "[kokkos_utils]")
   test_parallel_reduce<Real,15,false> ();
 }
 
+
+template<typename Scalar, bool Serialize, int TotalSize, int VectorSize>
+void test_view_reduction(const Scalar a=Scalar(0.0), const int begin=0, const int end=TotalSize)
+{
+  using Device = ekat::DefaultDevice;
+  using MemberType = typename ekat::KokkosTypes<Device>::MemberType;
+  using ExeSpace = typename ekat::KokkosTypes<Device>::ExeSpace;
+  
+  using PackType = ekat::Pack<Scalar, VectorSize>;
+  using ViewType = Kokkos::View<PackType*,ExeSpace>;
+
+  const int view_length = ekat::npack<PackType>(TotalSize);
+
+  // Each entry is given by data(k)[p] = 1/(k*Pack::n+p+1)
+  Scalar serial_result = Scalar(a);
+  ViewType data("data", view_length);
+  const auto data_h = Kokkos::create_mirror_view(data);
+  auto raw = data_h.data(); 
+  for (int k = 0; k < view_length; ++k) {
+    for (int p = 0; p < VectorSize; ++p) {
+      const int scalar_index = k*VectorSize+p;
+      if (scalar_index >= TotalSize) {
+        // represents pack garbage
+        raw[k][p] = ekat::ScalarTraits<Scalar>::invalid();
+      } else {
+        const Scalar val = 1.0/(k*VectorSize+p+1);
+        raw[k][p] = val;
+
+        if (scalar_index >= begin && scalar_index < end) {
+          serial_result += val;
+        }
+      }
+    }
+  }
+  Kokkos::deep_copy(data, data_h);
+
+  Kokkos::View<Scalar*> results ("results", 1);
+  const auto results_h = Kokkos::create_mirror_view(results);
+
+  // parallel_for over 1 team, i.e. call view_reduction once
+  const auto policy =
+    ekat::ExeSpaceUtils<Serialize,ExeSpace>::get_default_team_policy(1, view_length);
+  Kokkos::parallel_for(policy, KOKKOS_LAMBDA(const MemberType& team) {
+    Scalar team_result = Scalar(a);
+
+    ekat::ExeSpaceUtils<Serialize,ExeSpace>::view_reduction(team, begin, end, data, team_result);
```

Comment:
You should test the lambda InputProvider case, as well. Can just wrap data in a lambda. In trial set 0, use the raw data view; in trial set 1, use the lambda wrapper. In general, in unit tests for ekat, _every_ possible input type must be tested.